qiskit_ibm_runtime/base_primitive.py

```diff
@@ -183,6 +183,12 @@ def _run_primitive(self, primitive_inputs: Dict, user_kwargs: Dict) -> RuntimeJo
             combined["resilience_level"] = Options._DEFAULT_RESILIENCE_LEVEL
 
         self._validate_options(combined)
+        if combined["resilience_level"] == 2:
+            if not combined["resilience"]["noise_factors"]:
+                combined["resilience"]["noise_factors"] = (1, 3, 5)
+            if not combined["resilience"]["extrapolator"]:
+                combined["resilience"]["extrapolator"] = "LinearExtrapolator"
+
```

Comment:
Maybe it's better to have this in a helper method? Not sure if this is the best base to check for this

Reply:
I agree with your comment here. I suggest moving all the default setting into a separate method (`set_default_options()`).